rest/stats_context_test.go

```diff
@@ -62,3 +66,41 @@ func TestDescriptionPopulation(t *testing.T) {
 	// assert on a HELP description
 	assert.Contains(t, string(bodyString), `HELP sgw_cache_high_seq_stable The highest contiguous sequence number that has been cached.`)
 }
+
+func TestMemoryProfile(t *testing.T) {
+	stats := statsContext{}
+
+	outputDir := t.TempDir()
+	ctx := base.TestCtx(t)
+
+	// collect single profile
+	startTime := "01"
+	require.NoError(t, stats.collectMemoryProfile(ctx, outputDir, startTime))
+	require.Equal(t, []string{"pprof_heap_high_01.pb.gz"}, getFilenames(t, outputDir))
+
+	// collect enough profiles to trigger rotation
+	expectedFilenames := make([]string, 0, 10)
+	for i := 2; i < 12; i++ {
+		// reset heap profile time time to ensure we create new heap profiles
+		stats.lastHeapProfile = time.Time{}
+		expectedFilenames = append(expectedFilenames, fmt.Sprintf("pprof_heap_high_%02d.pb.gz", i))
+		require.NoError(t, stats.collectMemoryProfile(ctx, outputDir, fmt.Sprintf("%02d", i)))
+	}
+	slices.Sort(expectedFilenames)
+	require.Equal(t, expectedFilenames, getFilenames(t, outputDir))
```

Comment:
For future you could use require.ElementsMatch() instead of having to sort the slice.